src/config.rs

```diff
@@ -0,0 +1,77 @@
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[derive(TypeAbi, TopEncode, TopDecode)]
+pub struct Offer<M: ManagedTypeApi> {
+    pub creator: ManagedAddress<M>,
+    pub offered_payment: EsdtTokenPayment<M>,
+    pub accepted_payment: EsdtTokenPayment<M>,
+    pub opt_accepted_address: Option<ManagedAddress<M>>,
+}
+
+impl<M: ManagedTypeApi> Offer<M> {
+    pub fn new(
+        creator: ManagedAddress<M>,
+        offered_payment: EsdtTokenPayment<M>,
+        accepted_payment: EsdtTokenPayment<M>,
+        opt_accepted_address: Option<ManagedAddress<M>>,
```

Comment:
I don't think we need the address to be optional. An escrow is defined between two parties that want to transact with each other, so by definition, the contract is correct while enforcing the address

Reply:
Updated.